File: test_plugin/src/main/kotlin/dev/kotx/test_plugin/TestPlugin.java
package dev.kotx.test_plugin;

import dev.kotx.flylib.FlyLibKt;
import dev.kotx.flylib.command.Command;
import org.bukkit.event.block.BlockBreakEvent;
import org.bukkit.plugin.java.JavaPlugin;

public class TestPlugin extends JavaPlugin {
    @Override
    public void onEnable() {
        FlyLibKt.flyLib(this, flyLib -> {
            flyLib.command(new ExplodeCommand());
            flyLib.listen(BlockBreakEvent.class, event -> {
                //block break event logic
            });
        });
    }
}

class ExplodeCommand extends Command {
    public ExplodeCommand() {
        super("explode");

        usage(usage -> {
            usage.entityArgument("targets");
            usage.integerArgument("power", 1, 10);
            usage.selectionArgument("mode", "one", "two");

            usage.executes(context -> {
                context.message("You executed explode command!");
            });
        });
    }
}

File: test_plugin/src/main/kotlin/dev/kotx/test_plugin/TestPlugin.kt
/*
 * Copyright (c) 2021 kotx__.
 * Twitter: https://twitter.com/kotx__
 */

package dev.kotx.test_plugin

import dev.kotx.flylib.command.Command
import dev.kotx.flylib.flyLib
import org.bukkit.plugin.java.JavaPlugin

class TestPlugin : JavaPlugin() {
    init {
        flyLib {
            command()
        }
    }
}

class ExplodeCommand : Command("explode") {
    init {
        usage {
            entityArgument("targets")
            integerArgument("power", min = 1, max = 10)
            selectionArgument("mode", "")
        }
    }
}